### ISSUE TITLE
[Core] Allow modification of Netty HttpClient inside NettyAsyncHttpClientBuilder

### ISSUE THREAD

**Issue by mikeharder** [Azure/azure-sdk-for-java#5891]
There is no way to modify the Netty `HttpClient` inside `NettyAsyncHttpClientBuilder` (e.g. to allow untrusted SSL certs).  Here's the code which creates and returns the Netty `HttpClient`.

https://github.com/Azure/azure-sdk-for-java/blob/0478750cd9ac2354992d5ed2cc7c1ae198e25e2f/sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/NettyAsyncHttpClientBuilder.java#L46-L92

A few options:

1. Add a constructor to `NettyAsyncHttpClientBuilder` which accepts a Netty `HttpClient` instance.
2. Add a method to modify the Netty `HttpClient` instance after it's been created.

A workaround is to use private reflection:

```java
HttpClient httpClient = HttpClient.createDefault();

Field nettyClientField = httpClient.getClass().getDeclaredField("nettyClient");
nettyClientField.setAccessible(true);

reactor.netty.http.client.HttpClient nettyClient =
        (reactor.netty.http.client.HttpClient) nettyClientField.get(httpClient);

// modify nettyClient

nettyClientField.set(httpClient, nettyClient);
```

CC: @JonathanGiles

**Comment by JonathanGiles** [comment]
@anuchandy had a proof of concept on how this was to be implemented. My preference was to tackle this after the GA.

**Comment by mikeharder** [comment]
@JonathanGiles: Before or after GA is your call.  Perf tests are unblocked via the private reflection hack.

**Comment by alzimmermsft** [comment]
@anuchandy a constructor exists that allows passing of a Netty HttpClient, can this issue be closed?

**Comment by alzimmermsft** [comment]
Closing as this issue has been fixed by #6179 which added support to base the `NettyAsyncHttpClient` on a pre-configured Netty `HttpClient`.
